app/models/manageiq/providers/openstack/infra_manager.rb

```diff
@@ -83,4 +83,31 @@ def supports_authentication?(authtype)
   def self.event_monitor_class
     ManageIQ::Providers::Openstack::InfraManager::EventCatcher
   end
+
+  def verify_credentials(auth_type = nil, options = {})
+    auth_type ||= 'default'
+
+    raise MiqException::MiqHostError, "No credentials defined" if missing_credentials?(auth_type)
+
+    options[:auth_type] = auth_type
+    case auth_type.to_s
+    when 'default' then     verify_api_credentials(options)
+    when 'amqp' then        verify_amqp_credentials(options)
+    when 'ssh_keypair' then verify_ssh_keypair_credentials(options)
+    else;                   raise "Invalid OpenStack Authentication Type: #{auth_type.inspect}"
+    end
+  end
+
+  def required_credential_fields(type)
+    case type.to_s
+    when 'ssh_keypair' then [:userid, :auth_key]
+    else                    [:userid, :password]
+    end
+  end
+
+  def verify_ssh_keypair_credentials(_options)
+    hosts.sort_by(&:ems_cluster_id).slice_when { |i, j| i.ems_cluster_id != j.ems_cluster_id }.map { |c| c.find { |h| h.power_state == 'on' } }.all? { |h| h.verify_credentials('ssh_keypair') }
```

Comment:
I am grouping over clusters b/c I want only test on one running host of each cluster, not on each running host - see `{ |c| c.find { |h| h.power_state == 'on' } }`

I talked with @lsmola and @tzumainn about how to define this validation. We decided that validating over all running hosts would be slow so I validate only over one running host of each cluster. If you think that this should be changed feel free to tell me.